### PR TITLE
Fix setup issues for Roslyn insertion

### LINKED ISSUE #1654: FSharp Binaries installed reference wrong versions
During VS install, NGen fails because loaded binaries reference wrong versions of their dependencies:

`ExecNetFx:  Failed to load dependency System.ValueTuple of assembly FSharp.Compiler, Version=4.4.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a because of the following error : The system cannot find the file specified. (Exception from HRESULT: 0x80070002)`

`ExecNetFx:  Failed to load dependency FSharp.Core of assembly FSharp.Data.TypeProviders, Version=4.3.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a because of the following error : The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)`

`ExecNetFx:  1>Error compiling C:\Program Files (x86)\Microsoft SDKs\F#\4.1\Framework\v4.0\fsc.exe: The image file is not compatible with the version of Ngen you're running.  Use 32bit Ngen for 32bit assemblies, and 64bit Ngen for 64bit assemblies. (Exception from HRESULT: 0x80131F07)`

`ExecNetFx:  1>Error compiling C:\Program Files (x86)\Microsoft SDKs\F#\4.1\Framework\v4.0\fsi.exe: The image file is not compatible with the version of Ngen you're running.  Use 32bit Ngen for 32bit assemblies, and 64bit Ngen for 64bit assemblies. (Exception from HRESULT: 0x80131F07)`

### LINKED ISSUE #1655: Upgrade to SwixBuild plugin version 1.0.71

### PR DESCRIPTION
**Pull request by OmarTawfik**
Fixes #1655 
Fixes #1654

- Provided an application path for NGENed binaries, and fixed app.config redirections for TypeProviders and ValueTuples dependencies.
- Fixed NGENing fsc/fsi as x86 applications.

@Microsoft/fsharp-compiler

### CODE REVIEW

File: vsintegration/Vsix/VisualFSharpDesktop/Source.extension.vsixmanifest [hunk 1]
```diff
@@ -59,7 +59,4 @@
     <Asset Type="Microsoft.VisualStudio.MefComponent" d:Source="Project" d:ProjectName="FSharp.Editor" Path="|FSharp.Editor|" AssemblyName="|FSharp.Editor;AssemblyName|" />
     <Asset Type="Microsoft.VisualStudio.Analyzer" d:Source="Project" d:ProjectName="FSharp.Editor" Path="|FSharp.Editor|" />
   </Assets>
-  <Prerequisites>
```
Comment (Pilchie) [inline vsintegration/Vsix/VisualFSharpDesktop/Source.extension.vsixmanifest]: Why are we removing these?

File: setup/packages.config [hunk 2]
```diff
@@ -7,10 +7,11 @@
...
   <package id="MicroBuild.Core" version="0.2.0" />
   <package id="MicroBuild.Core.Sentinel" version="1.0.0" />
-  <package id="MicroBuild.Plugins.SwixBuild" version="1.0.60" />
+  <package id="MicroBuild.Plugins.SwixBuild" version="1.0.71" />
   <package id="WiX.Toolset.2015" version="3.10.0.1503" />
   <package id="Microsoft.VisualFSharp.Core.Redist" version="1.0.0" />
   <package id="Microsoft.VisualStudio.Shell.14.0" version="14.3.25407" targetFramework="net46" />
   <package id="Microsoft.VisualFSharp.Msbuild.15.0" version="1.0.1" />
   <package id="System.ValueTuple" version="4.0.0-rc3-24212-01" />
+  <package id="Microsoft.VisualFSharp.Type.Providers.Redist" version="1.0.0" />
```
Comment (Pilchie) [inline setup/packages.config]: I don't see this as part of the change?

Comment (Pilchie) [inline setup/packages.config]: Is this because you moved it here from vs-integration?

File: src/fsharp/Fsc/fsc.exe.config [hunk 3]
```diff
@@ -13,6 +13,15 @@
...
       </dependentAssembly>
+      <dependentAssembly>
+        <assemblyIdentity
+          name="System.ValueTuple"
+          publicKeyToken="b03f5f7f11d50a3a"
+          culture="neutral"/>
+        <bindingRedirect
+          oldVersion="2.0.0.0-4.4.1.0"
+          newVersion="4.0.0.0"/>
+      </dependentAssembly>
```
Comment (Pilchie) [inline src/fsharp/Fsc/fsc.exe.config]: Is this related to the bug above?

File: setup/packages.config (hunk 2 above)
Comment (OmarTawfik) [inline setup/packages.config]: Moved. These are not needed in vsintegration. Only used in setup.

File: src/fsharp/Fsc/fsc.exe.config (hunk 3 above)
Comment (OmarTawfik) [inline src/fsharp/Fsc/fsc.exe.config]: Yes. Needed so that NGEN can load the correct dependencies.

File: src/fsharp/Fsc/fsc.exe.config [hunk 4]
```diff
@@ -13,6 +13,15 @@
           oldVersion="2.0.0.0-4.4.1.0"
           newVersion="4.4.1.0"/>
       </dependentAssembly>
+      <dependentAssembly>
+        <assemblyIdentity
+          name="System.ValueTuple"
+          publicKeyToken="b03f5f7f11d50a3a"
+          culture="neutral"/>
+        <bindingRedirect
+          oldVersion="2.0.0.0-4.4.1.0"
```
Comment (KevinRansom) [inline src/fsharp/Fsc/fsc.exe.config]: I don't think value tuple needs this binding redirect.  In the future it may need one.  but not right now.

File: src/fsharp/fsi/fsiAnyCPU.exe.config [hunk 5]
```diff
@@ -12,6 +12,15 @@
           oldVersion="2.0.0.0-4.4.1.0"
           newVersion="4.4.1.0"/>
       </dependentAssembly>
+      <dependentAssembly>
```
Comment (KevinRansom) [inline src/fsharp/fsi/fsiAnyCPU.exe.config]: Value tuple shouldn't need a binding redirect.

Comment (OmarTawfik) [inline src/fsharp/fsi/fsiAnyCPU.exe.config]: @KevinRansom  We are referencing version 4.0.0.0, so fsc shouldn't be affected by this change. BUT this is needed for NGEN. Do you have concerns for adding this redirection here?

Comment (OmarTawfik) [inline src/fsharp/fsi/fsiAnyCPU.exe.config]: Discussed offline. We can get rid of the dependency.
